compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -737,8 +736,7 @@ fn receiver_is_dispatchable<'tcx>(
     // use a bogus type parameter to mimic a forall(U) query using u32::MAX for now.
     // FIXME(mikeyhew) this is a total hack. Once object_safe_for_dispatch is stabilized, we can
     // replace this with `dyn Trait`
-    let unsized_self_ty: Ty<'tcx> =
-        Ty::new_param(tcx, u32::MAX, Symbol::intern("RustaceansAreAwesome"));
+    let unsized_self_ty: Ty<'tcx> = Ty::new(tcx, ty::Param(ty::ParamTy::dummy()));
```

Comment:
`ty::ParamTy::dummy()` now automatically implements all `Auto` traits, as I noted below, right? I'm suspicious that `receiver_is_dispatchable` is now subtly unsound.

Reply:
Same as above.